samtranslator/model/api/http_api_generator.py

```diff
@@ -586,6 +585,27 @@ def _construct_stage(self):
 
         return stage
 
+    def _set_description(self):
+        """Sets Description to DefinitionBody"""
+        if not self.description:
+            return
+
+        if not self.definition_body:
+            raise InvalidResourceException(
+                self.logical_id,
+                "Description works only with inline OpenApi specified in the 'DefinitionBody' property.",
+            )
+        if self.definition_body.get("info", {}).get("description"):
```

Comment:
is description value with empty string accepted?

Reply:
`description` can be set as empty string in the OpenAPI definition, but APIGateway won't update the API Description. On the contrary, if `description` is not empty in the OpenAPI definition, APIGateway will update the API Description with the value of `description`. 
I think the idea is - if `description` is an empty string and the `Description` property is set in the SAM template, we probably want to use the value of `Description` to overwrite that of `description` in the OpenAPi definition. Does that make sense?